### ISSUE TITLE
SchemaNode should default to Mapping?

### ISSUE THREAD

**Issue by tisdall** [Pylons/colander#230]
The docs say the following about [colander.SchemaNode](http://docs.pylonsproject.org/projects/colander/en/latest/api.html#colander.SchemaNode):
> typ: The ‘type’ for this node. It should be an instance of a class that implements the colander.interfaces.Type interface. **If typ is not passed, it defaults to colander.Mapping().**

This is currently not true:
```
>>> import colander
>>> x = colander.SchemaNode()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/sites/metrics_dev/env/lib/python3.4/site-packages/colander/__init__.py", line 1817, in __init__
    self.typ = self.schema_type()
  File "/sites/metrics_dev/env/lib/python3.4/site-packages/colander/__init__.py", line 1833, in schema_type
    'Schema node construction without a typ argument or '
NotImplementedError: Schema node construction without a typ argument or a schema_type() callable present on the node class 
>>> 
```

So, I can provide a PR, but should the docs be changed or the code?

**Comment by tisdall** [comment]
Based on https://github.com/Pylons/colander/pull/162#issuecomment-34585152 and https://github.com/Pylons/colander/pull/162#issuecomment-40632100  I'm leaning towards just changing the documentation.

**Comment by mmerickel** [comment]
You should update the docs such that `SchemaNode` and `Schema` are not a mapping. Then you should fork and fix #162 such that `MappingSchema`, `TupleSchema`, `SequenceSchema`, etc all properly do not require a  `typ` parameter as its first argument with the requested tests.

Step 2 is that I think it should be possible to pass `typ` and `typ_opts` to the `__init__` to override the type, or to leave the type as the default but pass extra opts to it upon construction. However let's focus on the first part first.

**Comment by tisdall** [comment]
@mmerickel - okay, so is #232 sufficient for the doc update?

**Comment by tisdall** [comment]
> Then you should fork and fix #162 such that MappingSchema, TupleSchema, SequenceSchema, etc all properly do not require a typ parameter as its first argument with the requested tests.

You mean #168, right?

**Comment by mmerickel** [comment]
@tisdall Yeah that's what I meant.

**Comment by mmerickel** [comment]
Closed via #232.
